lang/java/pom.xml

```diff
@@ -361,6 +361,7 @@
                       <pluginExecutionFilter>
                         <groupId>com.diffplug.spotless</groupId>
                         <artifactId>spotless-maven-plugin</artifactId>
+                        <version>${spotless-maven-plugin.version}</version>
```

Comment:
```suggestion
                        <versionRange>${spotless-maven-plugin.version}</versionRange>
```
Hello!  Good catch, but I couldn't get it to work with `<version>`, did you mean `<versionRange>` ?

Reply:
Truth be told, this builds fine from the command line, but my Eclipse is unhappy about it, and when Eclipse is unhappy, I am unhappy. :)

Check out the error message again:
```
Missing parameter for pluginExecutionFilter. groupId, artifactId, versionRange and goals must be specificed, but found: groupId = 'com.diffplug.spotless'
artifactId = 'spotless-maven-plugin'
versionRange = 'null'
goals = '[check]'
```

It requires 5 fields, but only 4 are listed in the "but found" section.  The missing one is `artifiactId`.  'null' perhaps could be a valid value for `versionRange `

Reply:
Also note that CI does work for JDK8 and JDK 11, so it is building with the current change.

Reply:
I'm glad someone is looking out for working in Eclipse :D  Happy devs make happy software!

I don't use Eclipse as often, so I tested with Eclipse 2020-03 and still couldn't successfully build after your change.  

I just did **File** / **Import** / **Existing Maven Projects** and pointed to the top-level `pom.xml` and got a bunch of these errors:

`Cannot parse lifecycle mapping metadata for maven project MavenProject: org.apache.avro:avro-parent:1.11.0-SNAPSHOT @ /home/rskraba/working/github/avro/lang/java/pom.xml Cause: Unrecognised tag: 'version' (position: START_TAG seen ...</artifactId>\n        <version>... @8:18) 	pom.xml	/avro-ipc-netty	line 1`

It works if the XML tag added to the pom is `<versionRange>`, which is why I asked.  Am I doing something wrong while checking this?

Reply:
Hey Ryan,

Really appreciate the help on this.  It's quite annoying.

I just pulled down the latest changes to the master branch and did "Import [Maven] Project" and when it's doing it's initial build, I get this error message.  Nothing special to it.  May be a Eclipse version issue. 

if you add both a `<version>` and `<versionRange>` tag, does that build?  That may be the resolution to make all versions happy.

Reply:
I stuck the `version` tag in here, deleted the project, imported it again, and it came up without the annoying error.

Reply:
I suggest just pushing this change to trunk, and if someone is using a different version of eclipse, we can deal with that in a new Jira/PR

Reply:
![image](https://user-images.githubusercontent.com/7744819/84166446-31b40b80-aa75-11ea-9b39-0e64752441f3.png)

I still have the same error on Eclipse 2019-06 -- if `<version>` (or both) is present, the Maven Project Build Lifecycle Mapping fails!  This is Fedora 32 by the way.

In any case, the maven import to eclipse is definitely broken as it is, and slightly less broken (even for me!) after the PR.  Let's merge it, or get someone else to try?  Either way, I have more confidence in your setup than mine!